staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise/interface.go

```diff
@@ -16,27 +16,85 @@ limitations under the License.
 
 package promise
 
-// Mutable is a variable that is initially not set and can be set one
-// or more times (unlike a traditional "promise", which can be written
-// only once).
-type Mutable interface {
+// This file defines interfaces for promsies and futures and related
+// things.
 
+// Readable is a variable that is initially not set and later becomes
+// set.  Some instances may be set to multiple values in series.  A
+// Readable for a variable that can eventually get just one value is
+// commonly known as a "future".
+type Readable interface {
+	// Get reads the current value of this variable.  If this variable
+	// is not set yet then this call blocks until this variable gets a
+	// value.
+	Get() interface{}
+
+	// IsSet returns immediately with an indication of whether this
+	// variable has been set.
+	IsSet() bool
+}
+
+// ReadableLocked is a Readable whose implementation is protected by a lock
+type LockingReadable interface {
```

Comment:
From the func names below, it seems Locked is better than Locking

Reply:
"locked" is a state that a mutex may or may not be in; entered via Mutex::Lock() and exited by Mutex::Unlock().  I am following the convention here that ending a method name in "Locked" signals that the caller must already have locked the mutex.  I chose "LockingReadable" rather than "LockedReadable" because the Readable's mutex is not always locked, rather I am trying to signal that locking is part of how the Readable works.